apps/extensions_manager/lib/extensions_manager/model_extension.ex

```diff
@@ -0,0 +1,75 @@
+defmodule ExtensionsManager.ModelExtension do
+  defmacro __using__(_opts) do
+    quote do
+      Module.register_attribute(__MODULE__, :schema_changes, accumulate: true)
+      Module.register_attribute(__MODULE__, :method_block, accumulate: true)
+
+      import ExtensionsManager.ModelExtension, only: [add_to_schema: 4, add_to_schema: 3, include_method: 1]
+      @before_compile ExtensionsManager.ModelExtension
+
+      defmacro __using__(_opts) do
```

Comment:
@mgwidmann Project is in very early stages. It will evolve where `Nectar` and all other extensions will be phoenix projects which can be downloaded as hex packages.

Nectar and extensions can be grown and develop in isolations. Nectar being maintained by one team and extensions probably by many different teams where no one team will own all the extensions and Nectar Project.

So, Nectar should provide a way to extensions to hook back and grow independently.

Changes in extensions should not result / demand a change in Nectar Project but it could work otherwise, wherein extensions can adapt to change as per Nectar, if need some specific functionality which is provided by Nectar later.

Not very sure, whether above answers your question or not :( :disappointed: 

We are in process of releasing the Blog Series which will provide in-depth details of nature of project and how it would be used.

We hope, it should make things more clear :blush: